### ISSUE TITLE
Feature request: playlist support

### ISSUE THREAD

**Issue by GyrosGeier** [dirkjanm/firefox-send-to-xbmc#12]
I have a bunch of videos saved to playlists, which gives me a different page layout, so the extension does no longer work.

It'd be great if the alternate layout for playlist mode could be recognized, and/or the entire playlist sent to Kodi.

**Comment by dirkjanm** [comment]
Are you talking about a YouTube playlist?

**Comment by GyrosGeier** [comment]
Yes, I am.

**Comment by dirkjanm** [comment]
The issue with videos in playlists was solved in 7ddb0e4a1739a0be94bc0184ce9b009dc53f54a6
Sending full playlists is not supported for now but maybe in a future version this can be added.

**Comment by GyrosGeier** [comment]
Thanks a lot, makes things a lot easier.

**Comment by theBear** [comment]
It would be really great, if you could add "add to playlist" to the option, especially for music, narrations or even podcasts. I just wanna right-click the mp3 (e.g.) ein enqueue items to the kodi-playlist. Is it possible?

**Comment by bloody76** [comment]
Hi,

I made a fork with something which looks like to playlist support if you want. I don't know if it's what you are looking for but for the moment it:
- can add files to a queue on xbmc,
- can add mixes from YouTube to a queue on xbmc.

I changed the behavior of the play button, first it synchronizes with the xbmc server, then it adds the file to the playlist and then asks xbmc to play the last file in the playlist.

For the moment it's supporting only one playlist (so only one server also), but the server part is easy to change.

Also, I changed a little bit the design in order to have 3 buttons, feel free to change it, I made it simple because I am not very fond of css.

Can you tell me if I should do a pull request or if you want me to review some things first ? Also, do anyone have any idea of doing a better playlist support ?

**Comment by dirkjanm** [comment]
Could you fix the whitespaces? It's hard to see the changes since GitHub considers everything changed. If you use an editor which supports the .editorconfig it would be even easier.

It looks like a good start for playlist support, I think it would be good to create a separate branch for it, if you do a PR against that branch then I can test it and work some more on it (possibly adding support for multiple servers).

**Comment by bloody76** [comment]
Hi,

Hum I am not sure if I can fix the whitespaces, what do you mean by that ? You want me to put the indent level to 2 spaces ? (in my commit it's 4).

Ok for the branch, I also made a new commit in order to add a button to YouTube playlists. It adds the whole playlist and play the first one.

**Comment by dirkjanm** [comment]
Yes, the project uses 2 spaces as ident size. I've created the new branch.

**Comment by bluezed** [comment]
Just wondering... is there any progress on this feature or can the branch be used safely for now? 

I'd really like that feature too... either queueing videos or even better sending a whole playlist would be great.

As a reference... On my Android phone I have Yatse installed and through that I can queue and send playlists to Kodi directly. Some similar functionality would be perfect.

Thank you very much :+1:

**Comment by dirkjanm** [comment]
@bluezed the branch is still in development, the feature works partially but still has a lot of errors with adding/replacing videos in the playlist.
